azure-pipeline-codeql.yml

```diff
@@ -0,0 +1,155 @@
+variables:
+  - name: _TeamName
+    value: DotNetCore
+  - group: SDL_Settings
+
+trigger: none
+
+schedules:
+  - cron: 0 12 * * 1
+    displayName: Weekly Monday CodeQL/Semmle run
+    branches:
+      include:
+      - main
+    always: true
+
+stages:
+- stage: CodeQL
+  displayName: CodeQL
+
+  jobs:
+    - job: CSharp
+      timeoutInMinutes: 90
+      pool: 
+        name: NetCore1ESPool-Internal
+        demands: ImageOverride -equals Build.Server.Amd64.VS2019
+      displayName: "CodeQL Scan"
+      steps:
+
+      # Guardian requirements
+      - task: NuGetToolInstaller@1
+        displayName: 'Install NuGet.exe'
+
+      - task: NuGetAuthenticate@0
+        displayName: 'Authenticate NuGet'
+        inputs:
+          nuGetServiceConnections: GuardianConnect
+  
+      - pwsh: |
+          . $(Build.SourcesDirectory)\eng\CodeQL.ps1
+          $guardianCliLocation = Install-Gdn -Path $(Build.SourcesDirectory)\.artifacts
+          Write-Host "##vso[task.setvariable variable=GuardianCliLocation]$guardianCliLocation"
+        displayName: Install Guardian
+
+      # Project requirements
+      - task: UseDotNet@2
+        displayName: Install Correct .NET Version
+        inputs:
+          useGlobalJson: true
+
+      - task: UseDotNet@2
+        displayName: Install .NET Version 3.1
+        inputs:
+          version: 3.1.x
+
+      - task: NodeTool@0
+        displayName: 'Install Node'
+        inputs:
+          versionSpec: '12.x'
+
+      - task: NuGetCommand@2
+        displayName: Restore Packages
+        inputs:
+          command: restore
+          solution: "**/*.sln"
+          feedstoUse: config
+
+      - pwsh: eng\set-version-parameters.ps1
+        displayName: Calculate release version variables
+        
+      - pwsh: |
+          [xml]$manifest = Get-Content src\Maestro\MaestroApplication\ApplicationPackageRoot\ApplicationManifest.xml
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Policies']").RemoveAll()
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Principals']").RemoveAll()
+          $manifest.Save("src\Maestro\MaestroApplication\ApplicationPackageRoot\ApplicationManifest.xml")
+          git diff
+        displayName: Remove Service Fabric RunAsPolicy from MaestroApplication
+
+      - pwsh: |
```

Comment:
Maybe it's obvious and I'm just unfamiliar with CodeQL but why are you modifying XML files as part of CodeQL build?

Reply:
This exists in the regular azure-pipelines.yaml and is involved in building the source. I just duplicated it here to ensure CodeQL is close to the "real" build.